lib/pure/uri.nim

```diff
@@ -311,6 +356,46 @@ proc `$`*(u: Uri): string =
     result.add("#")
     result.add(u.anchor)
 
+proc parseQuery*(query: string): OrderedTableRef[string, string] = 
```

Comment:
Create a new ``Query`` type. Make it a ``distinct OrderedTableRef[string, string]``. Set it as this procedure's return type.